ServiceWebsite/ServiceWebsite/ClientApp/src/app/modules/individual-journey/services/submit.service.ts

```diff
@@ -0,0 +1,75 @@
+import { Injectable } from '@angular/core';
+import { HearingSuitabilityAnswer } from 'src/app/services/clients/api-client';
+import { IndividualModelMapper } from './individual-model-mapper';
+import { SuitabilityService } from './suitability.service';
+import { SuitabilityAnswer, HasAccessToCamera } from '../../base-journey/participant-suitability.model';
+import { IndividualSuitabilityModel } from '../individual-suitability.model';
+
+@Injectable()
+export class SubmitService {
+
+  constructor(private suitabilityService: SuitabilityService) { }
+
+  async submit(model: IndividualSuitabilityModel) {
+    const answers: HearingSuitabilityAnswer[] = new IndividualModelMapper().mapToRequest(model);
+    await this.suitabilityService.updateSuitabilityAnswers(model.hearing.id, answers);
+  }
+
+  isDropOffPoint(step: number, model: IndividualSuitabilityModel): boolean {
```

Comment:
Remember the part of the talk on coding, was it Monday, where there was a part about separating reads/writes.

This method name doesn't imply that the submission is actually taking place which can lead to some nasty issues.

Could you break it in two parts? One to just check if drop off should be done and one to do the actual drop off?
You may need to have an internal method to check which drop off point you're on and call this twice but I think it's better than opening this up with mixing submit&dropoff check

Reply:
Especially since you already have the `updateSubmitModel` step separate already.